src/util/createInstantSearch.js

```diff
@@ -75,19 +143,40 @@ export const createInstantSearch = instantSearchOptions => {
       warn('The result of `getState()` needs to be passed to `hydrate()`.');
       return;
     }
-    const { lastResults } = instantSearchState;
-    search.searchParameters = lastResults._state;
+    const { lastResults, derived } = instantSearchState;
```

Comment:
lastResults is actually duplicate with derived, so we probably should get rid of it?